rn/Teacher/src/modules/pages/edit/PageEdit.js

```diff
@@ -142,51 +147,54 @@ export class PageEdit extends Component<Props, State> {
                 onFocus={this._scrollToRCE}
               />
             </View>
-
-            <FormLabel>{i18n('Details')}</FormLabel>
-            { !this.state.isFrontPage &&
-              <RowWithSwitch
-                title={i18n('Publish')}
-                border='bottom'
-                value={this.state.published}
-                onValueChange={this.handlePublishedChange}
-                testID='pages.edit.published.row'
-                identifier='pages.edit.published.switch'
-              />
-            }
-            { !(page && page.isFrontPage) && this.state.published &&
-              <RowWithSwitch
-                title={i18n('Set as Front Page')}
-                border='both'
-                value={this.state.isFrontPage}
-                onValueChange={this.handleIsFrontPageChange}
-                testID='pages.edit.front_page.row'
-                identifier='pages.edit.front_page.switch'
-              />
-            }
-            <RowWithDetail
-              title={i18n('Can Edit')}
-              detailSelected={this.state.editingRolesPickerShown}
-              detail={editingRoles()[editingRole]}
-              disclosureIndicator
-              border='bottom'
-              onPress={this.toggleEditingRoles}
-              testID='pages.edit.editing_roles.row'
-            />
-            { this.state.editingRolesPickerShown &&
-              <PickerIOS
-                selectedValue={editingRole}
-                onValueChange={this.handleEditingRolesChange}
-                testID='pages.edit.editing_roles.picker'
-              >
-                {Object.keys(editingRoles()).map(key => (
-                  <PickerItem
-                    key={key}
-                    value={key}
-                    label={editingRoles()[key]}
+            {isTeacher() &&
+              <React.Fragment>
```

Comment:
Interesting. What is `React.Fragment`?

Reply:
`React.Fragment` is a component that essentially just wraps other components. So in this case when we want to conditionally render a couple of views that are all siblings, without `React.Fragment` we have 3 options.
 - We conditionally render each sibling individually which isn't great. 
 - We wrap them in a `View` and conditionally render that, but then we have an extra `View` hanging around that isn't really needed
 - We conditionally render an array of views. But with this then you have to provide a key for each element which isn't great.

So instead we use `React.Fragment` which is a wrapping component that renders nothing. Gives us the ability to conditionally render a bunch of sibling components without the extra `View`.